int32 robot_id
Task[] tasks